pkg/resources/opensearchrepository/opensearchrepository.go

```diff
@@ -0,0 +1,100 @@
+package opensearchrepository
+
+import (
+	"context"
+	"time"
+
+	"github.com/cisco-open/operator-tools/pkg/reconciler"
+	loggingv1beta1 "github.com/rancher/opni/apis/logging/v1beta1"
+	"github.com/rancher/opni/pkg/util/meta"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+	opensearchv1 "opensearch.opster.io/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Reconciler struct {
+	reconciler.ResourceReconciler
+	client      client.Client
+	respository *loggingv1beta1.OpensearchRepository
```

Comment:
In the future, do we plan to support other types of repositories? Like using plain shared file system (shared file system as a snapshot repository, add it to opensearch.yml)